OneCommunityGlobal/HighestGoodNetworkApp
# Vishala Phase 2 Update Materials Single and Bulk

# Description
This PR is about **updating single material** from material list page and **updating materials in bulk** from update materials page.
Or Implements # (WBS) 3.3.5

## Related PRS (if any):
Related Backend PR : https://github.com/OneCommunityGlobal/HGNRest/pull/606

## Main changes explained:
- Added Update Material Folder with multiple files

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. on the backend , do `npm run build` and `npm start`
4. log as admin user
5. Test **Materials List page** : http://localhost:3000/bmdashboard/materials
6. Please test for list view , project based listing of materials and open/close functionality of edit/view modals
7. **Test Update Material one at a time**  : http://localhost:3000/bmdashboard/materials
8. Click the update icon under updates to open the pop up window
9. Play with input values and units to test validations. 
10. Any value entered on Qty used/wasted should be added to used/wasted values and subtracted from available
11. Click submit to check if the values have been updated correctly in materials list page.
12. Open `view` near the update icon to view the history of the update made.
13. To test Bulk Material Update , go to http://localhost:3000/bmdashboard/updateMaterials
14. Test Select projects and Select date fields
15. Play with input values and units to test validations
16. Any value entered on Qty used/wasted should be added to used/wasted values and subtracted from available
17.Cancel should cancel the entered values and Submit should submit the data and you can view the update changes in materials list page.

Note : As it is a fresh implementation, I constructed it based on phase 2 documentation, wireframes, input gathered from fellow team members, and incorporating some of my own insights. Hence , Do conduct a thorough testing to ensure optimal user-friendliness.

## Screenshots or videos of changes:
[Untitled_ Nov 15,2023 5_04 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/f1176a2c-b7b0-4ca2-800c-923e15fbe0f8)
